phylogen.md

```diff
@@ -0,0 +1,441 @@
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+```{currentmodule} tskit
+```
+
+(sec_phylogen)=
+
+# `Tskit` for phylogenetics
+
+`Tskit`, the tree sequence toolkit, can be used as an efficient library for very large evolutionary trees. `Tskit` makes it easy to deal with trees with millions of
+tips, as in the example below:
+
+```{code-cell}
+:"tags": ["hide-input"]
+import tskit
+```
+
+```{code-cell}
+%%time
+
+num_tips = 1_000_000
+big_tree = tskit.Tree.generate_comb(num_tips);
+print("Tree sequence takes up", big_tree.tree_sequence.nbytes / 1024**2, "Mb")
+print(f"Generating a 'comb' (pectinate) tree of {num_tips} tips took:")
+```
+
+:::{todo}
+Display the million tip tree in condensed form, when
+https://github.com/tskit-dev/tskit/issues/2372#issuecomment-1298518380 and
+https://github.com/tskit-dev/tskit/issues/2628 are solved
+:::
+
+Calculations on these huge trees can be very efficient:
+
+```{code-cell}
+%%time
+
+traversed_nodes = 0
+for u in big_tree.nodes(order="postorder"):
+    traversed_nodes += 1
+print("Postorder traversal through {traversed_nodes} took:")
+```
+
+```{code-cell}
+%%time
+
+b1_index = big_tree.b1_index()
+print(f"B1 balance index is {b1_index}. Calculation took:")
+```
+
+We can also read trees efficiently, e.g. in Newick format:
+
+:::{todo}
+Add example of fast reading of a newick file, once
+https://github.com/tskit-dev/tskit/issues/2187 is solved. E.g. we could say
+> For example, we can read in a newick format tree of XXX tips in XXX secs. In practice,
+> this means when reading in a large tree you are mainly limited by disk access speeds
+:::
+
+```{code-cell}
+import tsconvert  # used for reading tree sequences from different formats
+
+# example code reading in a large file, timed
+
+# example of reading from a string (smaller tree) into a tree sequence
+# Todo: add sequence length: https://github.com/tskit-dev/tsconvert/issues/40
+ts = tsconvert.from_newick("(A:6,((B:1,C:1):2,(D:2,E:2):1):3);")
+```
+
+The "succinct tree sequence" format used by `tskit` can also store mutations
+(and optionally a reference genome) along with the tree(s). This results in a
+single unified representation of large genomic datasets, storing trees,
+sequence data and metadata in a single efficient structure. Examples are given
+in the  {ref}`end of this page<sec_phylogen_unified_structure>`.
+
+As the name suggests, a tree sequence can also store and analyse a sequence of
+trees along a genome (i.e. a "phylogenetic network"). This is necessary to
+account for recombination between lineages, and may be important even when looking at
+species-level phylogenies due to the effects of hybridization and incomplete lineage
+sorting. An overview, and links to further details are given
+{ref}`sec_phylogen_multiple_trees`
+
+## Hints for phylogeneticists
+
+Unlike other phylogenetic libraries, `tskit` is designed to efficiently store not just 
+single trees, but entire , and focusses primarily on dealing with 
```

Comment:
typo? missing word after "entire"?

Reply:
Well spotted, thanks! I have a few more additions too. I'll add that to them and push as a separate commit so it's easy to see.